src/net_processing.cpp

```diff
@@ -2605,6 +2609,39 @@ bool SendMessages(CNode* pto, CConnman& connman, std::atomic<bool>& interruptMsg
                 pto->fDisconnect = true;
             }
         }
+        // Check for headers sync timeouts
+        if (state.fSyncStarted && state.nHeadersSyncTimeout < std::numeric_limits<int64_t>::max()) {
+            // Detect whether this is a stalling initial-headers-sync peer
+            if (pindexBestHeader->GetBlockTime() <= GetAdjustedTime() - 24*60*60) {
```

Comment:
I believe this timeout logic is connected to `nMaxTipAge` value and should be adjusted just like we did on line 2399 earlier